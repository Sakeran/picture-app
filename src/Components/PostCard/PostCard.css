.PostCard-main {
  width: 200px;
  padding-top: 1em;
  margin-bottom: 1em;
}

.PostCard-main:hover {
  transform: scale(1.2);
  transition: transform 0.4s ease-in-out;
}

.PostCard-img {
  max-width: 90%;
  height: auto;
  display: block;
  margin: 0 auto;
  border-radius: 1rem;
  margin-bottom: 0.2rem;
}

.PostCard-stats {
  text-align: center;
}
