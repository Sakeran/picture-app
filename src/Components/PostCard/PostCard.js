import React from 'react';

import './PostCard.css';

const renderImage = (link, alt) => (
  <img className="PostCard-img" src={link} alt={alt} />
)

const renderYoutube = (id) => (
  <strong>Youtube Placeholder</strong>
);

const PostCard = (props) => (
  <div className="PostCard-main">
    <div className="PostCard-stats">
      {props.post.type === 'image' ? renderImage(props.post.image, props.post.title) : renderYoutube(props.post.youtubeID)}
      <strong> Post Stats </strong>
    </div>
  </div>
);

export default PostCard;
