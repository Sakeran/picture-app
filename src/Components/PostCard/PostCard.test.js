import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PostCard from './PostCard';

test('PostCard element renders correctly', () => {
  const post = {
    type: 'image',
    image: 'http://localhost:3001/exampleImg.png',
    title: 'test card',
    youtubeID: null
  };
  const component = shallow(<PostCard post={post} />);
  expect(component.find('img')).toHaveLength(1);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
