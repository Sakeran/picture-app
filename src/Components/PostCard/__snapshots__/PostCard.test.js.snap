// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard element renders correctly 1`] = `
<div
  className="PostCard-main"
>
  <div
    className="PostCard-stats"
  >
    <img
      alt="test card"
      className="PostCard-img"
      src="http://localhost:3001/exampleImg.png"
    />
    <strong>
       Post Stats 
    </strong>
  </div>
</div>
`;
